lib/types.js

```diff
@@ -2339,7 +2339,7 @@ RecordType.prototype._update = function (resolver, type, opts) {
       } else {
         resolvers[name].push(fieldResolver);
       }
-      innerArgs.push(field.name);
+      innerArgs.push('_' + field.name);
```

Comment:
To avoid prefixing the underscore in two different places, can we add it when creating `fieldResolver`'s `name` just above? We can then switch this line to use `fieldResolver.name` and the code below should do the right thing without changes.